### PR TITLE
refactors syslog and kafka URL validation to obs API

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR:
* refactors syslog and kafka URL validation to obs API

### Links
* https://issues.redhat.com/browse/LOG-5823
* https://issues.redhat.com/browse/LOG-5824

cc @Clee2691 @cahartma @vparfonov

### CODE REVIEW

File: api/observability/v1/input_types.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ var (
 // +kubebuilder:validation:XValidation:rule="self.type != 'application' || has(self.application)", message="Additional type specific spec is required for the input type"
 // +kubebuilder:validation:XValidation:rule="self.type != 'infrastructure' || has(self.infrastructure)", message="Additional type specific spec is required for the input type"
 // +kubebuilder:validation:XValidation:rule="self.type != 'audit' || has(self.audit)", message="Additional type specific spec is required for the input type"
-// +kubebuilder:validation:XValidation:rule="self.type != 'receiver' || has(self.receiver)", message="Additional type specific spec is required for the input type"
+
```
Comment (vparfonov) [inline api/observability/v1/input_types.go:56]: Why do you remove it?

File: internal/validations/observability/outputs/validate_url_to_output_tls.go [hunk 2]
```diff
@@ -19,13 +19,13 @@ func validateURLAccordingToTLS(output obs.OutputSpec) (results []string) {
 	case obs.OutputTypeHTTP:
```
Comment (vparfonov) [inline internal/validations/observability/outputs/validate_url_to_output_tls.go:19]: Can we use URL for other output types too?

File: api/observability/v1/input_types.go (hunk 1 above)
Comment (jcantrill) [reply api/observability/v1/input_types.go:56]: accident i believe

File: internal/validations/observability/outputs/validate_url_to_output_tls.go (hunk 2 above)
Comment (jcantrill) [reply internal/validations/observability/outputs/validate_url_to_output_tls.go:19]: reverting URL and adding marker to all URLs

File: api/observability/v1/output_types.go [hunk 3]
```diff
@@ -606,11 +608,13 @@ type Kafka struct {
 	// first connection only. The collector's Kafka client fetches constantly an updated list
 	// from Kafka. These updates are not reconciled back to the collector configuration.
 	// If none provided the target URL from the OutputSpec is used as fallback.
-	//
 	// +kubebuilder:validation:Optional
-	Brokers []string `json:"brokers,omitempty"`
+	Brokers []URL `json:"brokers,omitempty"`
```
Comment (vparfonov) [inline api/observability/v1/output_types.go:612]: string ?

File: internal/generator/vector/output/kafka/kafka.go [hunk 4]
```diff
@@ -131,7 +133,7 @@ func Topics(o obs.OutputSpec) string {
 		return o.Kafka.Topic
 	}
 
-	url, _ := urlhelper.Parse(o.Kafka.URL)
+	url, _ := urlhelper.Parse(string(o.Kafka.URL))
```
Comment (vparfonov) [inline internal/generator/vector/output/kafka/kafka.go:136]: string() redundant here

File: test/framework/functional/output_syslog.go [hunk 5]
```diff
@@ -18,7 +18,7 @@ func (f *CollectorFunctionalFramework) AddSyslogOutput(b *runtime.PodBuilder, ou
 	log.V(2).Info("Adding syslog output", "name", output.Name)
 	name := strings.ToLower(output.Name)
 	var baseRsyslogConfig string
-	u, _ := url.Parse(output.Syslog.URL)
+	u, _ := url.Parse(string(output.Syslog.URL))
```
Comment (vparfonov) [inline test/framework/functional/output_syslog.go:21]: string() redundant here

File: internal/generator/vector/output/syslog/syslog.go [hunk 6]
```diff
@@ -94,7 +94,7 @@ func New(id string, o obs.OutputSpec, inputs []string, secrets vectorhelpers.Sec
 			Debug(id, vectorhelpers.MakeInputs(inputs...)),
 		}
 	}
-	u, _ := url.Parse(o.Syslog.URL)
+	u, _ := url.Parse(string(o.Syslog.URL))
```
Comment (vparfonov) [inline internal/generator/vector/output/syslog/syslog.go:97]: string() redundant here

**Pull request merged**